content/blog/tune-0-1-1/index.Rmd

```diff
@@ -0,0 +1,255 @@
+```
+
+This tells a completely different story than the previous version where the parameters were in their natural units. 
+
+The regular grid results are much better and tell a cleaner story: 
+
+```{r ap-reg-new, dev = 'svg', dev.args = list(bg = "transparent"), fig.align='center', fig.height=4, fig.width=5.5}
+autoplot(regular_grid, metric = "rmse") + 
+  ggtitle("new method, regular grid")
+```
+
+Extra arguments can be passed when a numeric grouping column is used and these are given to `format()`. To avoid scientific notation:
+
+```{r ap-reg-new-new, dev = 'svg', dev.args = list(bg = "transparent"), fig.align='center', fig.height=4, fig.width=5.5}
+autoplot(regular_grid, metric = "rmse", digits = 3, scientific = FALSE) + 
+  ggtitle("Formatting for coloring column")
+```
+
+## A `ggplot2` `coord` for plotting observed and predicted values
+
+One helpful visualization of the fit of a regression model is to plot the true outcome value against the predictions. These _should_ be on the same scale. Let's look at such a plot: 
+
+```{r obs-pred-old, dev = 'svg', dev.args = list(bg = "transparent"), fig.align='center', fig.height=4}
+best_values <- select_best(regular_grid, metric = "rmse")
+best_values
+
+holdout_predictions <- 
+  regular_grid %>% 
+  collect_predictions(parameters = best_values)
+holdout_predictions
+
+ggplot(holdout_predictions, aes(x = ridership, y = .pred)) + 
+  geom_abline(lty = 2) + 
+  geom_point(alpha = 0.3)
+```
+
+This can be very helpful but there are a few possible improvements. The new version of tune has `coord_obs_pred()` that produces a square plot with the same axes: 
+
+```{r obs-pred-new, dev = 'svg', dev.args = list(bg = "transparent"), eval = TRUE, fig.align='center', fig.height=4}
+ggplot(holdout_predictions, aes(x = ridership, y = .pred)) + 
+  geom_abline(lty = 2) + 
+  geom_point(alpha = 0.3) + 
+  coord_obs_pred()
+```
+
+
+## Tuning engine parameters
+
+[Bruna Wundervald](https://github.com/brunaw) (from Maynooth University) have a [great presentation](https://brunaw.com/slides/rladies-helsinki/talk.html#1) that used tidymodels packages. She ran into the problem that, if you wanted to tune parameters that we specific to the engine, you have to go to a lot of trouble to do so. This did work on an earlier version of tune and we fixed this issue. One feature in this version of tune, along with the new 0.0.8 version of the dials package, is that we have added dials `parameter` objects for every parameter that users are likely to tune with the existing engines that we support (this was not as difficult as it sounds).
+
+To demonstrate, we'll use the times series data above but optimize the `ranger` parameters that Bruna is interested in. 
+
+Since parsnip has a pre-defined list of models and engines, we've gone ahead and setup the infrastructure for tuning most engine-specific values. In the example above, what would happen if we tried to tune the two regularization parameters? 
```

Comment:
```suggestion
Since parsnip has a pre-defined list of models and engines, we've gone ahead and set up the infrastructure for tuning most engine-specific values. For example, in the above example we could tune two regularization parameters specific to ranger.
```